snuba/clusters/cluster.py

```diff
@@ -137,37 +141,46 @@ def __init__(
         super().__init__(storage_sets)
         if not single_node:
             assert cluster_name
-        self.__host = host
-        self.__port = port
+
+        self.__query_node = ClickhouseNode(host, port)
         self.__http_port = http_port
         self.__single_node = single_node
         self.__cluster_name = cluster_name
         self.__distributed_cluster_name = distributed_cluster_name
         self.__reader: Optional[Reader[SqlQuery]] = None
         self.__connection_cache: MutableMapping[
-            ClickhouseClientSettings, ClickhousePool
+            Tuple[ClickhouseNode, ClickhouseClientSettings], ClickhousePool
         ] = {}
 
     def __str__(self) -> str:
-        return f"{self.__host}:{self.__port}"
+        return str(self.__query_node)
 
     def get_connection(
```

Comment:
I'd consider just having a separate method to make it more clear what a method should be used for.
Like `get_query_connection` and `get_node_connection`

Reply:
Sure